cmake/EthCompilerSettings.cmake

```diff
@@ -35,6 +35,7 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
 	add_compile_options(-Wall)
 	add_compile_options(-Wextra)
 	add_compile_options(-Werror)
+	add_compile_options(-Weffc++)
```

Comment:
This is interesting flag I was not aware of.

Reply:
I was also not aware of, but sadly it produces to much noise (at least for gcc). It may be useful in some situations.

Reply:
From GCC docs, this flag seems to be oblate because it's based on quite old C++ recommendations pre C++11. I think using clang-tidy would give much better results.

Reply:
Yep thats true! I thought about adding clang-tidy to the build-system.